libvast/src/system/read_query.cpp

```diff
@@ -114,13 +114,21 @@ read_query(const invocation& inv, std::string_view file_option,
                                          *fname));
     return read_query(*fname);
   }
-  if (has_query_cli) {
-    if (has_query_stdin)
+  if (has_query_stdin && has_query_cli) {
+    auto query = read_query(std::cin);
+    // As a special case we ignore stdin unless we get an actual query
+    // from it, since empirically this is often connected to the remnants
+    // of some long-forgotten file descriptor by some distant parent
+    // process when running in any kind of automated environment, e.g. CI.
+    if (!query || query->empty())
+      has_query_stdin = false;
```

Comment:
One case is missing here: if the query is empty and we do not have a query on stdin, we should still export all. E.g.,

```
echo '' | vast export json
```

Reply:
I'm not sure about this, actually. Is "empty query" the same as "no query?"

Reply:
Did you read the context in Slack? We're facing this issue in CI where there's always something on stdin, so we get a lot of problems and false positives. Not doing this additional check will just cause the same trouble for other invocations.